src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/TestClasses/MethodOverloadTestExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Extensions.Logging.Generators.Tests.TestClasses
+{
+    internal static partial class OverloadTestExtensions
+    {
+        [LoggerMessage(EventId = 0, Level = LogLevel.Trace, Message = "M0{p0}")]
+        public static partial void M0(ILogger logger, int p0);
+
+        [LoggerMessage(EventId = 1, Level = LogLevel.Trace, Message = "M0{p0}")]
+        public static partial void M0(ILogger logger, string p0);
+
+
+        [LoggerMessage(EventId = 2, EventName = "M1Custom", Level = LogLevel.Trace, Message = "M1{p0}")]
```

Comment:
Aren't the first two `M0` enough? are you testing something new in the methods `M1` below? If not, then can perhaps remove.

Reply:
My thinking for this was to verify that custom event names are honoured for each individual overload of a logging method. As the implementation is now, it doesn't do a lot, but if the implementation was to change in future as to how EventNames are processed in the parser, then this test would highlight that issue.